packages/ember-application/lib/system/engine-instance.js

```diff
@@ -194,9 +194,25 @@ const EngineInstance = EmberObject.extend(RegistryProxyMixin, ContainerProxyMixi
 
     this.inject('view', '_environment', '-environment:main');
     this.inject('route', '_environment', '-environment:main');
+  },
+
+  [FACTORY_FOR](fullName, options) {
```

Comment:
Lets make this part of the `ContainerProxyMixin`, this would simplify issues in `ember-test-helpers` land and actually keep the logic in the correct place (currently `.lookup` and `._lookupFactory` are done in `ContainerProxyMixin`).